app/Models/Course.php

```diff
@@ -51,4 +51,59 @@ public function scopeOtherCourses($query)
     {
         return $query->skip(config('variable.skip_courses_home'))->take(config('variable.num_courses_home'));
     }
+
+    public function getTotalLearnersAttribute()
+    {
+        return $this->users()->count();
+    }
+
+    public function getTotalLessonsAttribute()
+    {
+        return $this->lessons()->count();
+    }
+
+    public function getTotalTimesAttribute()
+    {
+        return $this->lessons()->sum(config('variable.sum'));
+    }
+
+    public function scopeSearch($query, $data)
+    {
+        if (isset($data['keyword'])) {
+            $query->where('name', 'LIKE', "%".$data['keyword']."%")
+                    ->orWhere('description', 'LIKE', "%".$data['keyword']."%");
+        }
+
+            (isset($data['created_time']) && $data['created_time'] == config('variable.sort_by_newest')) ? $query->orderBy('created_at', config('variable.orderby_direction')) : $query->orderBy('created_at');
+
+        if (isset($data['teachers'])) {
+            $query->whereHas('teachers', function ($query) use ($data) {
```

Comment:
giải thích cho anh câu query này bằng lời và câu query trong mysql là gì

Reply:
Câu query trong SQL: select * from courses where exists (select * from users inner join course_teacher on users.id = course_teacher.user_id where courses.id = course_teacher.course_id and user_id in (1) and users.deleted_at is null) and courses.deleted_at is null

Theo em hiểu câu query này là: Lấy tất cả các courses có id trùng với id trong bảng course_teacher và có giáo viên có id=1 sao cho các courses không trùng nhau